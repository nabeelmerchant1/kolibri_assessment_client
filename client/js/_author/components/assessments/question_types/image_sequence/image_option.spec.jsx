import React          from 'react';
import { shallow }    from 'enzyme';
import renderer       from 'react-test-renderer';
import ImageOption    from './image_option';

describe('image_option component', () => {
  let props;
  let result;
  let activatedChoice;
  let updatedChoice;
  let inputUpdate;

  beforeEach(() => {
    inputUpdate = false;
    activatedChoice = false;
    updatedChoice = false;
    props = {
      deleteChoice: () => {},
      updateChoice: () => {
        updatedChoice = true;
        inputUpdate = true;
      },
      activateChoice: () => { activatedChoice = true; },
      id: '7',
      text: 'sample_text',
      labelText: '',
      order: 2,
      numChoices: 3,
      localizedStrings: () => {},
    };
    result = shallow(<ImageOption {...props} />);
  });

  it('renders a snapshot', () => {
    const tree = renderer.create(<ImageOption text={props.text} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls activateChoice function', () => {
    expect(activatedChoice).toBeFalsy();
    result.find('.au-c-image-sequence-answer').simulate('click');
    expect(activatedChoice).toBeTruthy();
  });

  it('calls updateChoice function on select change', () => {
    expect(updatedChoice).toBeFalsy();
    result.find('select').simulate('change', { target: { value: 4 } });
    expect(updatedChoice).toBeTruthy();
  });

  it('should not render the label input', () => {
    expect(inputUpdate).toBeFalsy();
    expect(result.find('input').length).toBe(0);
  });
});
