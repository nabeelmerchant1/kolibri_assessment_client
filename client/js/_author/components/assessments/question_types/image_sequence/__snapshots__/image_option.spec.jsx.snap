// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image_option component renders a snapshot 1`] = `
<div
  className="au-c-image-sequence-answer is-active"
  onClick={[Function]}
  tabIndex="0"
>
  <div
    className="au-c-image-sequence-answer__top"
  >
    <div
      className="au-c-dropdown au-c-dropdown--tiny "
    >
      <label
        htmlFor="image_option_order"
      />
      <select
        id="image_option_order"
        name=""
        onChange={[Function]}
        value={undefined}
      >
        <option
          value={null}
        >
          N/A
        </option>
        <option
          value={1}
        >
          1
        </option>
      </select>
    </div>
    <button
      className="au-c-answer--delete au-u-right"
      onClick={[Function]}
    >
      <i
        className="material-icons"
      >
        close
      </i>
    </button>
  </div>
  <div
    className="au-c-input au-c-input-label--left"
  >
    <div
      className="au-c-input__contain"
    >
      <div
        className="au-c-fake-input-smaller"
      />
    </div>
  </div>
  <div
    className="au-c-image-sequence-answer__image"
  >
    <img
      alt=""
      src="sample_text"
    />
  </div>
</div>
`;
